basic/http/collect-user.js

```diff
@@ -25,13 +28,36 @@ class User {
 };
 
 const user = new User();
+
 router.get( '/users', async ctx => {
-  ctx.body = {data:user.data.map((e)=>{
+  ctx.body = {
+    data:user.data.map((e)=>{
       return {id:e.id,name:e.name}; 
     })
   };
+});
 
+router.post('/users',async ctx => {
+  
+  user.data.push(ctx.request.body);
+  
+  ctx.body = {
+    data:user.data.map((e)=>{
+      return {id:e.id,name:e.name}; 
+    })
+  };
 });
+
+router.patch('/users/:id',async ctx => {  
+  user.data[R.indexOf(ctx.params.id, user.data)] = ctx.request.body;
```

Comment:
To update the array of objects you can just use the `for..of`  functions.
Please read this [article](https://bobbyhadz.com/blog/javascript-update-property-of-object-in-array)